chia/plotting/manager.py

```diff
@@ -254,154 +253,145 @@ def refresh_batch(self, plot_paths: List[Path], plot_directories: Set[Path]) ->
+                    cache_entry = CacheEntry(pool_public_key, pool_contract_puzzle_hash, plot_public_key)
+                    self.cache.update(prover.get_id(), cache_entry)
 
-                    if file_path in self.failed_to_open_filenames:
-                        del self.failed_to_open_filenames[file_path]
-
-                except Exception as e:
-                    tb = traceback.format_exc()
-                    log.error(f"Failed to open file {file_path}. {e} {tb}")
-                    self.failed_to_open_filenames[file_path] = int(time.time())
-                    return new_provers
-                log.info(f"Found plot {file_path} of size {new_provers[file_path].prover.get_size()}")
-
-                if self.show_memo:
-                    plot_memo: bytes32
-                    if pool_contract_puzzle_hash is None:
-                        plot_memo = stream_plot_info_pk(pool_public_key, farmer_public_key, local_master_sk)
+                with self.plot_filename_paths_lock:
+                    if file_path.name not in self.plot_filename_paths:
+                        self.plot_filename_paths[file_path.name] = (str(Path(prover.get_filename()).parent), set())
                     else:
-                        plot_memo = stream_plot_info_ph(pool_contract_puzzle_hash, farmer_public_key, local_master_sk)
-                    plot_memo_str: str = plot_memo.hex()
-                    log.info(f"Memo: {plot_memo_str}")
-
-                return new_provers
-            return new_provers
-
-        def reduce_function(x: Dict, y: Dict) -> Dict:
-            return {**x, **y}
+                        self.plot_filename_paths[file_path.name][1].add(str(Path(prover.get_filename()).parent))
+                    if len(self.plot_filename_paths[file_path.name][1]) > 0:
+                        log.warning(
+                            f"Have multiple copies of the plot {file_path} in "
+                            f"{self.plot_filename_paths[file_path.name][1]}."
+                        )
+                        return None
+
+                new_plot_info: PlotInfo = PlotInfo(
+                    prover,
+                    cache_entry.pool_public_key,
+                    cache_entry.pool_contract_puzzle_hash,
+                    cache_entry.plot_public_key,
+                    stat_info.st_size,
+                    stat_info.st_mtime,
+                )
+
+                with counter_lock:
+                    result.loaded_plots += 1
+                    result.loaded_size += stat_info.st_size
+
+                if file_path in self.failed_to_open_filenames:
+                    del self.failed_to_open_filenames[file_path]
+
+            except Exception as e:
+                tb = traceback.format_exc()
+                log.error(f"Failed to open file {file_path}. {e} {tb}")
+                self.failed_to_open_filenames[file_path] = int(time.time())
+                return None
```

Comment:
technically this is a semantic change. If an exception is thrown after line 368 (362 in the new version):
```
new_provers[file_path] = PlotInfo(
```

previously it would return that `PlotInfo` object. It may not be possible for that to happen (as the code looks right now) and either way, it would probably be incorrect to return that new object in case of an exception.